lib/SPIRV/SPIRVReader.h

```diff
@@ -220,6 +220,44 @@ class SPIRVToLLVM {
   // OpenCL function always has NoUnwind attribute.
   // Change this if it is no longer true.
   bool isFuncNoUnwind() const { return true; }
+
+  bool isFuncReadNone(const std::string& name) const {
+    if (BuiltInConstFunc.find(name) != BuiltInFunc.end())
+      return true;
+    else
+      return false;
+  }
+
+  std::set<std::string> BuiltInConstFunc{
```

Comment:
Is it possible to make it `static const` and move it to SPIRVReader.cpp?

Probably it's worth trying to use `llvm::StringSet` here. It has more efficient memory handling. Though I'm not sure it keeps the elements ordered.